avail-server/src/main/java/com/avail/server/io/WebSocketAdapter.kt

```diff
@@ -86,14 +102,33 @@ import kotlin.experimental.xor
  *   The socket address of the listener.
  * @param serverAuthority
  *   The server's authority, e.g., the host name of this node.
+ * @param heartbeatFailureThreshold
+ *   The number of consecutive times the `heartbeatTimeout` is allowed to be
+ *   reached before disconnecting the client.
+ * @param heartbeatInterval
+ *   The time in milliseconds between each [Heartbeat] request made by the
+ *   server to the client after receiving a `Heartbeat` from the client.
+ * @param heartbeatTimeout
+ *   The amount of time in milliseconds the heartbeat will fail if a heartbeat
+ *   is not received from the client by the server.
+ * @param onChannelCloseAction
+ *   A function that accepts a [DisconnectReason] and the underlying
+ *   [channel][AbstractTransportChannel] and answers `Unit` that is to be
+ *   called when the input channel is closed.
  * @throws IOException
  *   If the [server socket][AsynchronousServerSocketChannel] could not be
  *   opened.
  */
 class WebSocketAdapter @Throws(IOException::class) constructor(
 	override val server: AvailServer,
 	internal val adapterAddress: InetSocketAddress,
-	internal val serverAuthority: String)
+	internal val serverAuthority: String,
+	private val heartbeatFailureThreshold: Int = 3,
+	private val heartbeatInterval: Long = 12000,
+	private val heartbeatTimeout: Long = 15000,
+	override val onChannelCloseAction:
+		(DisconnectReason, AbstractTransportChannel<AsynchronousSocketChannel>)
+			-> Unit = { _, _ -> /* Do nothing */})
```

Comment:
Great use of default arguments. If it gets much more onerous, we might want to introduce a `WebSocketAdapterConfiguration`. I'll probably need to do that in order to introduce new features, but we can worry about that later.

Reply:
One small request here — please introduce named `const val` fields in the companion to deanonymize the magic numbers `3`, `12000`, and `15000`. The empty arity-2 block is fine anonymous and in situ, however.